elyra/pipeline/registry.py

```diff
@@ -0,0 +1,483 @@
+
+        print("BODYY: " + repr(component_body))
+
+        # TODO Do we need to/should we pop these?
+        for element in ['runtime_image', 'env_vars', 'dependencies', 'outputs', 'include_subdirectories']:
+            component_parameters['current_parameters'].pop(element, None)
+
+        # Define new input group object
+        input_group_info = {
+            'id': "nodeInputControls",  # need to actually figure out the control id
+            'type': "controls",
+            'parameter_refs': []
+        }
+
+        inputs = component_body['inputs']
+        for input_object in inputs:
+            new_parameter, parameter_info = self.build_parameter(input_object, "input")
+
+            # TODO: Adjust this to return an empty value for whatever type the parameter is?
+            default_value = ""
+            if "default" in input_object:
+                default_value = input_object['default']
+
+            # Add to existing parameter list
+            component_parameters['parameters'].append(new_parameter)
+            component_parameters['current_parameters'][new_parameter['id']] = default_value
+
+            # Add to existing parameter info list
+            component_parameters['uihints']['parameter_info'].append(parameter_info)
+
+            # Add parameter to input group info
+            input_group_info['parameter_refs'].append(new_parameter['id'])
+
+        # Append input group info to parameter details
+        component_parameters['uihints']['group_info'][0]['group_info'].append(input_group_info)
+
+        # Define new output group object
+        output_group_info = {
+            'id': "nodeOutputControls",  # need to actually figure out the control id
+            'type': "controls",
+            'parameter_refs': []
+        }
+
+        outputs = component_body['outputs']
+        for output_object in outputs:
+            new_parameter, parameter_info = self.build_parameter(output_object, "output")
+
+            # TODO Adjust this to return an empty value for whatever type the parameter is?
+            default_value = ""
+
+            # Add to existing parameter list
+            component_parameters['parameters'].append(new_parameter)
+            component_parameters['current_parameters'][new_parameter['id']] = default_value
+
+            # Add to existing parameter info list
+            component_parameters['uihints']['parameter_info'].append(parameter_info)
+
+            # Add parameter to output group info
+            output_group_info['parameter_refs'].append(new_parameter['id'])
+
+        # Append output group info to parameter details
+        component_parameters['uihints']['group_info'][0]['group_info'].append(output_group_info)
+
+        return component_parameters
+
+    def build_parameter(self, obj, obj_type):
+        new_parameter = {}
+
+        new_parameter['id'] = f"elyra{obj_type}_{obj['name'].replace(' ', '_')}"
```

Comment:
Should we have a separator between `elyra` and the parameter type?  It might read easier.

Reply:
Oh, yeah, that is odd.  Could `obj_type` be made plural?  e.g., `elyra_inputs_input_data`, `elyra_inputs_packages_to_install`, `elyra_outputs_output_data`, etc.  This might better differentiate the "prefix" from the "parameter".

Reply:
I like that a lot!

Reply:
Now that the UI is integrated, I went with a slightly different approach. Turns out the id is only used internally, so I removed the `elyra_inputs` prefix from the inputs altogether, and only added `elyra_outputs` prefix if a prefix of the same name already exists in the parameter list.